### PR TITLE
fixed infinite while loop in cross_validation_wrapper

### PR DESCRIPTION
**Pull request by thevarunsharma**
I noticed that this problem was marked as TODO in the learning.py code, so I tried fixing it. I hope it will be working fine now i.e. no more forever running while loop.

### CODE REVIEW

**Comment by SiluPanda** [comment]
Hey, what I see you are doing is, converging the training error to 0. But in that case, it would just be overfit on the training set, loosing the generality. For example:, let's say we have 4 data sets and four labels. The model might take a polynomial of higher degree (greater than 4) and easily classify all four of them correctly, but that would perform horribly on the validation set and test set.

I hope I'm making sense to you.

**Comment by thevarunsharma** [comment]
Thanks for your suggestion @SiluPanda, but I want to clear thet after checkin convergence of training error, I am selecting the size which yielded the best val. error, so, it won't be choosing the overfitted solution.
Plus, please suggest a good threshold for train error, if close to 0 is not a good one.

**Comment by SiluPanda** [comment]
There are a lot of ways to add a threshold (not the optimal one, but a reasonably sub optimal).
There are a lot of discussions on this already. I believe, deviating from pseudocode should be with the concern of @norvig sir.

**Comment by thevarunsharma** [comment]
But the code is similar to the pseudocode provided here [https://github.com/aimacode/aima-pseudocode/blob/master/md/Cross-Validation-Wrapper.md](url)

**Comment by thevarunsharma** [comment]
but the code is similar to the pseudocode at https://github.com/aimacode/aima-pseudocode/blob/master/md/Cross-Validation-Wrapper.md

**Comment by SiluPanda** [comment]
Yes, it is similar. Let's wait for contributors to add their feedback.

**Comment by dmeoli** [comment]
Thanks @thevarunsharma. learning.py has since been refactored and cross_validation_wrapper reimplemented (the old TODO version this patched is gone), so this no longer applies. Closing as superseded — thanks!
